# HG changeset patch
# User Stephan Bosch <stephan@rename-it.nl>
# Date 1343690295 -7200
# Node ID c52a0c5613119fdae86b29a797d5d442c1e8d959
# Parent  e3ea0270985aa297e45e575001a22321b0b87524
Fix linkage with ld.gold.
Patch by Eray Aslan.

diff -r e3ea0270985a -r c52a0c561311 src/lib-sieve/Makefile.am
--- a/src/lib-sieve/Makefile.am	Mon Jul 30 10:14:42 2012 +0200
+++ b/src/lib-sieve/Makefile.am	Tue Jul 31 01:18:15 2012 +0200
@@ -70,8 +70,8 @@
 	$(extdir)/vnd.dovecot/duplicate/libsieve_ext_duplicate.la \
 	$(unfinished_plugins)
 
-libdovecot_sieve_la_DEPENDENCIES = $(plugins)
-libdovecot_sieve_la_LIBADD = $(plugins)
+libdovecot_sieve_la_DEPENDENCIES = $(plugins) $(LIBDOVECOT_LDA_DEPS)
+libdovecot_sieve_la_LIBADD = $(plugins) $(LIBDOVECOT_LDA)
 
 libdovecot_sieve_la_SOURCES = \
 	rfc2822.c \


# HG changeset patch
# User Stephan Bosch <stephan@rename-it.nl>
# Date 1342130270 -7200
# Node ID 6ceeb642123117ed8377f3b0d7d28171f790e911
# Parent  602d93069d3a1c93866178308bd4d03f20a15f73
ManageSieve: fixed segfault bug triggered by CHECKSCRIPT command.

diff -r 602d93069d3a -r 6ceeb6421231 src/lib-sievestorage/sieve-storage-save.c
--- a/src/lib-sievestorage/sieve-storage-save.c	Sat May 26 00:16:35 2012 +0200
+++ b/src/lib-sievestorage/sieve-storage-save.c	Thu Jul 12 23:57:50 2012 +0200
@@ -318,19 +318,20 @@
 bool sieve_storage_save_will_activate
 (struct sieve_save_context *ctx)
 {
-	const char *scriptname;
-	int ret = 0;
+	bool result = FALSE;
 
-	T_BEGIN {
+	if ( ctx->scriptname != NULL ) T_BEGIN {
+		const char *scriptname;
+		int ret;
+
 		ret = sieve_storage_get_active_scriptfile(ctx->storage, &scriptname);
-	
 		if ( ret > 0 ) {
 		 	/* Is the requested script active? */
-			ret = ( strcmp(ctx->scriptname, scriptname) == 0 ? 1 : 0 );
+			result = ( strcmp(ctx->scriptname, scriptname) == 0 );
 		}
 	} T_END;
 
-	return ret;
+	return result;
 }
 
 int sieve_storage_save_commit(struct sieve_save_context **ctx)

